For example

```
FAILED tests/models/wav2vec2/test_modeling_wav2vec2.py::Wav2Vec2ModelTest::test_mismatched_shapes_have_properly_initialized_weights - AssertionError: 0.4431610107421875 not found in [0.0, 1.0] : Parameter wav2vec2.masked_spec_embed of model <class 'transformers.models.wav2vec2.modeling_wav2vec2.Wav2Vec2ForSequenceClassification'> seems not properly initialized

FAILED tests/models/wav2vec2/test_modeling_wav2vec2.py::Wav2Vec2RobustModelTest::test_mismatched_shapes_have_properly_initialized_weights - AssertionError: 0.4431610107421875 not found in [0.0, 1.0] : Parameter wav2vec2.masked_spec_embed of model <class 'transformers.models.wav2vec2.modeling_wav2vec2.Wav2Vec2ForSequenceClassification'> seems not properly initialized


FAILED tests/models/clip/test_modeling_clip.py::CLIPForImageClassificationModelTest::test_mismatched_shapes_have_properly_initialized_weights - AssertionError: 0.0022945739328861237 not found in [0.0, 1.0] : Parameter classifier.weight of model <class 'transformers.models.clip.modeling_clip.CLIPForImageClassification'> seems not properly initialized

FAILED tests/models/regnet/test_modeling_regnet.py::RegNetModelTest::test_mismatched_shapes_have_properly_initialized_weights - AssertionError: 0.0011449280427768826 not found in [0.0, 1.0] : Parameter regnet.embedder.embedder.convolution.weight of model <class 'transformers.models.regnet.modeling_regnet.RegNetForImageClassification'> seems not properly initialized

```

Values like `0.4431610107421875`, `0.0022945739328861237` or `0.0011449280427768826` are all reasonable, despite it's neither `0.0` nor `1.0`.

--- a/tests/test_modeling_common.py
+++ b/tests/test_modeling_common.py
@@ -3211,7 +3211,7 @@ class ModelTesterMixin:
                     for name, param in new_model.named_parameters():
                         if param.requires_grad:
                             param_mean = ((param.data.mean() * 1e9).round() / 1e9).item()
-                            if not (is_special_classes and param in special_param_names):
+                            if not (is_special_classes and name in special_param_names):
                                 self.assertIn(
                                     param_mean,
                                     [0.0, 1.0],

--- a/tests/test_modeling_common.py
+++ b/tests/test_modeling_common.py
@@ -3179,14 +3179,19 @@ class ModelTesterMixin:
                 continue
 
             # TODO: ydshieh
-            if model_class.__name__ in [
-                "Wav2Vec2ForSequenceClassification",
+            is_special_classes = model_class.__name__ in [
+                "wav2vec2.masked_spec_embed",
                 "Wav2Vec2ForSequenceClassification",
                 "CLIPForImageClassification",
                 "RegNetForImageClassification",
                 "ResNetForImageClassification",
-            ]:
-                self.skipTest(reason="This test is currently failing for some models.")
+            ]
+            special_param_names = [
+                "wav2vec2.masked_spec_embed",
+                "classifier.weight",
+                "regnet.embedder.embedder.convolution.weight",
+                "resnet.embedder.embedder.convolution.weight",
+            ]
 
             with self.subTest(msg=f"Testing {model_class}"):
                 with tempfile.TemporaryDirectory() as tmp_dir:
@@ -3205,11 +3210,24 @@ class ModelTesterMixin:
 
                     for name, param in new_model.named_parameters():
                         if param.requires_grad:
-                            self.assertIn(
-                                ((param.data.mean() * 1e9).round() / 1e9).item(),
-                                [0.0, 1.0],
-                                msg=f"Parameter {name} of model {model_class} seems not properly initialized",
-                            )
+                            param_mean = ((param.data.mean() * 1e9).round() / 1e9).item()
+                            if not (is_special_classes and param in special_param_names):
+                                self.assertIn(
+                                    param_mean,
+                                    [0.0, 1.0],
+                                    msg=f"Parameter {name} of model {model_class} seems not properly initialized",
+                                )
+                            else:
+                                self.assertGreaterEqual(
+                                    param_mean,
+                                    0.0,
+                                    msg=f"Parameter {name} of model {model_class} seems not properly initialized",
+                                )
+                                self.assertLessEqual(
+                                    param_mean,
+                                    1.0,
+                                    msg=f"Parameter {name} of model {model_class} seems not properly initialized",
+                                )
 
     def test_matched_shapes_have_loaded_weights_when_some_mismatched_shapes_exist(self):
         # 1. Create a dummy class. Should have buffers as well? To make sure we test __init__

--- a/tests/test_modeling_common.py
+++ b/tests/test_modeling_common.py
@@ -3188,6 +3188,7 @@ class ModelTesterMixin:
             ]
             special_param_names = [
                 "wav2vec2.masked_spec_embed",
+                "wav2vec2.feature_extractor.conv_layers.*.conv.weight",
                 "classifier.weight",
                 "regnet.embedder.embedder.convolution.weight",
                 "resnet.embedder.embedder.convolution.weight",
@@ -3211,7 +3212,10 @@ class ModelTesterMixin:
                     for name, param in new_model.named_parameters():
                         if param.requires_grad:
                             param_mean = ((param.data.mean() * 1e9).round() / 1e9).item()
-                            if not (is_special_classes and name in special_param_names):
+                            if not (
+                                is_special_classes
+                                and any(len(re.findall(target, name)) > 0 for target in special_param_names)
+                            ):
                                 self.assertIn(
                                     param_mean,
                                     [0.0, 1.0],

